helm/kyverno-policies-observability/templates/PodMonitor.yaml

```diff
@@ -7,47 +7,60 @@ spec:
+        - resources:
+            kinds:
+            - monitoring.coreos.com/v1/PodMonitor
       mutate:
-        patchStrategicMerge:
-          spec:
-            podMetricsEndpoints:
-              - relabelings:
-                  # Pod Monitors are currently only supported on management clusters, so we can enforce the cluster_id and cluster_type.
-                - replacement: "{{ .Values.managementCluster.name }}"
-                  targetLabel: cluster_id
-                - replacement: "management_cluster"
-                  targetLabel: cluster_type
-                  # Since we only support management clusters, all resources are considered as highest service priority
-                - replacement: "highest"
-                  targetLabel: service_priority
-                - replacement: "{{ .Values.provider.kind }}"
-                  targetLabel: provider
-                - replacement: "{{ .Values.managementCluster.name }}"
-                  targetLabel: installation
-                - sourceLabels: [__meta_kubernetes_namespace]
-                  targetLabel: namespace
-                - sourceLabels: [__meta_kubernetes_pod_label_app_kubernetes_io_name]
-                  targetLabel: app
-                - sourceLabels: [__meta_kubernetes_pod_label_app_kubernetes_io_instance]
-                  targetLabel: instance
-                - sourceLabels: [__meta_kubernetes_pod_name]
-                  targetLabel: pod
-                - sourceLabels: [__meta_kubernetes_pod_container_name]
-                  targetLabel: container
-                - sourceLabels: [__meta_kubernetes_pod_node_name]
-                  targetLabel: node
-                - sourceLabels: [__meta_kubernetes_node_label_role]
-                  targetLabel: role
-                - replacement: "{{ .Values.managementCluster.customer }}"
-                  targetLabel: customer
-                ## We get the organization from the namespace (org-organization_name)
-                - replacement: "{{ `{{` }} request.namespace {{ `}}` }}" 
-                  targetLabel: organization
-                ## Then we remove the org prefix
-                - sourceLabels: [organization]
-                  regex: org-(.*)
-                  replacement: "${1}"
-                  targetLabel: organization
+        foreach:
+        - list: request.object.spec.podMetricsEndpoints
+          patchesJson6902: |-
+            # Pod Monitors are currently only supported on management clusters, so we can enforce the cluster_id and cluster_type.
+            - path: "/spec/podMetricsEndpoints/{{ `{{` }}elementIndex{{ `}}` }}/relabelings/-1"
+              op: add
+              value: {"replacement": "{{ .Values.managementCluster.name }}", "targetLabel": "cluster_id"}
+            - path: "/spec/podMetricsEndpoints/{{ `{{` }}elementIndex{{ `}}` }}/relabelings/-1"
+              op: add
+              value: {"replacement": "management_cluster", "targetLabel": "cluster_type"}
```

Comment:
It is quite hard to read those files, any way we could improve this?
I see we are adding multiple values, is there a way to combine those into one element maybe?
Maybe add a clear comment on what is the result produced in the end.

Reply:
Actually these files (helm templates) are generated from `policies/observability/` files by `make generate`.
The `policies/observability/` files are a bit easier to read because there's no escaping of `{{`, but yet not ideal.

I'll try one big json to be merged to the relabellings, may be easier to read.

Reply:
Didn't manage to have the exact same behaviour with a big json. I could replace all relabelings, but not append relabelings.
Asked Kyverno people for help, they did not have a better idea that the current syntax. :shrug: